### PR TITLE
feat(provider): migrate gitlab provider to ts

### PR DESCRIPTION
**Pull request by yoannfleurydev**
## ☕️ Reasoning

Migrate the GitLab provider to TypeScript. Better for type safety, avoid conflicts in projects that depends of NextAuth as mentioned in https://github.com/nextauthjs/next-auth/discussions/4728

## 🧢 Checklist

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

## 🎫 Affected issues

Related to a discussion: https://github.com/nextauthjs/next-auth/discussions/4728

Fixes: https://github.com/nextauthjs/next-auth/discussions/4728

## 📌 Resources

- [Contributing guidelines](./CONTRIBUTING.md)
- [Code of conduct](./CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Review by 0ubbe — APPROVED** [review]
Thanks 💚

File: packages/next-auth/src/providers/gitlab.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+export default function GitLab<P extends GitLabProfile>(
+  options: OAuthUserConfig<P>
+): OAuthConfig<P> {
+  return {
+    id: "gitlab",
+    name: "GitLab",
+    type: "oauth",
+    authorization: "https://gitlab.com/oauth/authorize?scope=read_user",
```
Comment (lluia) [inline packages/next-auth/src/providers/gitlab.ts:55]:
```suggestion
    authorization: {
      url: "https://gitlab.com/oauth/authorize",
      params: { scope: "read_user" },
    },
```
Just to be consistent with our latest OAuth provider API 👍🏽

Comment (yoannfleurydev) [reply packages/next-auth/src/providers/gitlab.ts:55]: Oh nice, didn't know about that one, commiting this in a minute.

File: packages/next-auth/src/providers/gitlab.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      url: "https://gitlab.com/oauth/authorize",
+      params: { scope: "read_user" },
+    },
+    token: "https://gitlab.com/oauth/token",
+    userinfo: "https://gitlab.com/api/v4/user",
+    checks: ["pkce", "state"],
+    profile(profile) {
+      return {
+        id: profile.id.toString(),
+        name: profile.username,
```
Comment (martin-fv) [inline packages/next-auth/src/providers/gitlab.ts:65]:
```suggestion
        name: profile.name ?? profile.username,
```

Comment (mshd) [inline packages/next-auth/src/providers/gitlab.ts:65]:
```suggestion
        name: profile.name ?? profile.username,
```
Idea: first use name and then fallback to username

**Review by mshd — APPROVED** [review]

File: packages/next-auth/src/providers/gitlab.ts (hunk 2 above)
Comment (yoannfleurydev) [reply packages/next-auth/src/providers/gitlab.ts:65]: You are right, then we will be consistent with GitHub provider for example. Doing the commit amend/push right now.

**Review by balazsorban44 — APPROVED** [review]
:metal: Thanks!

**Pull request merged**